lib/github_advisory_sync.rb

```diff
@@ -199,110 +206,122 @@ def ghsa_id
     # so a GitHub Security Advisory ID (ghsa_id) is used instead
     def primary_id
       return cve_id if cve_id
+
       ghsa_id
     end
 
     # return a date as a string like 2019-03-21.
     def published_day
-      return nil unless github_advisory_graphql_object["publishedAt"]
+      return unless advisory["publishedAt"]
 
-      pub_date = Date.parse(github_advisory_graphql_object["publishedAt"])
+      pub_date = Date.parse(advisory["publishedAt"])
       # pub_date.strftime("%Y-%m-%d")
       pub_date
     end
 
-    def package_names
-      github_advisory_graphql_object["vulnerabilities"]["nodes"].map{|v| v["package"]["name"]}.uniq
-    end
-
-    def rubysec_filenames
-      package_names.map do |package_name|
-        File.join("gems", package_name, "#{cve_id}.yml")
-      end
-    end
-
     def withdrawn?
-      !github_advisory_graphql_object["withdrawnAt"].nil?
+      !advisory["withdrawnAt"].nil?
     end
 
     def cvss
-      return "<FILL IN IF AVAILABLE>" if github_advisory_graphql_object["cvss"]["vectorString"].nil?
-      github_advisory_graphql_object["cvss"]["score"].to_f
+      return if advisory["cvss"]["vectorString"].nil?
+
+      advisory["cvss"]["score"].to_f
     end
 
     def external_reference
-      github_advisory_graphql_object["references"].find do |ref|
+      advisory["references"].each do |ref|
         next if ref["url"].start_with?("https://nvd.nist.gov/vuln/detail/")
-        ref["url"]
+
+        return ref["url"]
       end
     end
 
-    def vulnerabilities
-      github_advisory_graphql_object["vulnerabilities"]["nodes"]
+    def package_name
+      vulnerabilities.first["package"]["name"]
```

Comment:
in practice, this is fine but of note is that it's technically possible for one advisory to span many different package names.

Reply:
Do we expect that to ever happen? If so, we should probably support that... See also https://github.com/rubysec/ruby-advisory-db/issues/400

Reply:
Updated to allow for multiple package names

Reply:
@reedloden the schema changes enabling that shipped the same time we made this change https://github.com/rubysec/ruby-advisory-db/pull/479#discussion_r655432437

i think in practice it's just extremely rare 🤔 might not have happened yet in ruby land, i'd have to poke the API to find out :P.